### PR TITLE
Enhance the hyperlink text decoration option to enable/disable the default style handling (#1864)

### PR DESCRIPTION
**Pull request by speckyspooky**
- the feature includes the new switch option of the hyperlink text decoration option
- this feature is implemented with an adjustment of the ROM
- the change requires the new structure of the PerfectHash-values and the style constants (see according README.MD)
- the new style constants are created through the new processing of the PerfectHash according to the new attribute "text-decoration"

### CODE REVIEW

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/css/engine/StyleConstants.java [hunk 1]
```diff
@@ -93,146 +92,150 @@ public interface StyleConstants {
...
 
 	/** style property: letter spacing */
-	int STYLE_LETTER_SPACING = 69;
+	int STYLE_LETTER_SPACING = 70;
+
+	/** style property: text hyperlink style */
+	int STYLE_TEXT_HYPERLINK_STYLE = 40;
 
 	/** number (count) of style constants */
-	int NUMBER_OF_STYLE = 70;
```
Comment (wimjongman) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/css/engine/StyleConstants.java:237]: I don't understand why you renumber all constants. Are they not saved somewhere?

**Review by wimjongman — COMMENTED** [review]
This is a big change that I cannot fully grasp.

You say the ROM is changed. Does that not require an update of the ROM version number?

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/css/engine/StyleConstants.java (hunk 1 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/css/engine/StyleConstants.java:237]: The ROM version number can be changed. I will do it. 

The constants are changed due to the new indexing of PerfectHash. 
The new ordering of the Word-list with the new attribute crescendo the new Index values. 
This was done by the programme according to the README. MD

**Comment by speckyspooky** [comment]
I have added the new ROM-version number and renamed the new HTML-CSS-class.

**Comment by speckyspooky** [comment]
@wimjongman
Do you have further questions to this feature change before it is merged to the master?
I will be available to maintain the feature after merge & release.

**Review by wimjongman — APPROVED** [review]
Thanks, Thomas!

**Comment by wimjongman** [comment]
Are you going to create documentation for this feature?

**Comment by speckyspooky** [comment]
Yes, I can create a documentation.
What was your idea for documentation for this change?

**Comment by wimjongman** [comment]
Maybe on the documentation part of the site? https://github.com/eclipse-birt/birt-website/tree/main/docs

There are other sources as well. Maybe we can start a discussion about it.

**Pull request merged**